tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -808,6 +812,18 @@ Optional<UIdent> SwiftLangSupport::getUIDForDeclAttribute(const swift::DeclAttri
   return None;
 }
 
+UIdent SwiftLangSupport::getUIDForFormalAccessScope(const swift::AccessScope Scope) {
+  if (Scope.isPublic()) {
+    return EffectiveAccess_Public;
+  } else if (Scope.isInternal()) {
+    return EffectiveAccess_Internal;
+  } else if (Scope.isFileScope()) {
+    return EffectiveAccess_FilePrivate;
+  } else {
```

Comment:
To be defensive, I would still check `else if (Scope.isPrivate())`, then use `else { llvm_unreachable("Unsupported formal access scope") }`.

Nit: since `open` is technically an _access level_, but its scope would be public, what do you think of calling this feature just "effective access" or "effective access scope" instead of "effective access level"?